sdk/src/main/scala/com/horizen/account/storage/AccountStateMetadataStorageView.scala

```diff
@@ -0,0 +1,350 @@
+    }
+  }
+
+  override def hasCeased: Boolean = hasCeasedOpt.getOrElse(storage.get(ceasingStateKey).isPresent)
+
+  override def getHeight: Int = {
+    storage.get(heightKey).asScala.map(baw => Ints.fromByteArray(baw.data)).getOrElse(0)
+  }
+
+  private[horizen] def getAccountStateRootFromStorage: Option[Array[Byte]] = {
+    storage.get(accountStateRootKey).asScala.map(_.data)
+  }
+
+  override def getAccountStateRoot: Array[Byte] = {
+    accountStateRootOpt.orElse(getAccountStateRootFromStorage).getOrElse(DEFAULT_ACCOUNT_STATE_ROOT)
+  }
+
+  private[horizen] def getTransactionReceiptFromStorage(txHash: Array[Byte]): Option[EthereumReceipt] = {
+    storage.get(getReceiptKey(txHash)).asScala match {
+      case Some(serData) =>
+        val decodedReceipt: EthereumReceipt = EthereumReceiptSerializer.parseBytes(serData)
+        Some(decodedReceipt)
+
+      case None => None
+    }
+  }
+
+  override def getTransactionReceipt(txHash: Array[Byte]): Option[EthereumReceipt] = {
+    val bawTxHash = new ByteArrayWrapper(txHash)
+    receiptsOpt match {
+      case Some(receipts) =>
+        receipts.find(r => new ByteArrayWrapper(r.transactionHash) == bawTxHash)
+
+      case None => getTransactionReceiptFromStorage(txHash)
+    }
+  }
+
+  // put in memory cache and mark the entry as "dirty"
+  def updateWithdrawalEpochInfo(withdrawalEpochInfo: WithdrawalEpochInfo): Unit =
+    withdrawalEpochInfoOpt = Some(withdrawalEpochInfo)
+
+  def updateTopQualityCertificate(topQualityCertificate: WithdrawalEpochCertificate): Unit =
+    topQualityCertificateOpt = Some(topQualityCertificate)
+
+  def updateTransactionReceipts(receipts: Seq[EthereumReceipt]): Unit = {
+    receiptsOpt = Some(receipts)
+  }
+
+  def addFeePayment(blockFeeInfo: AccountBlockFeeInfo): Unit = {
+    blockFeeInfoOpt = Some(blockFeeInfo)
+  }
+
+  def updateConsensusEpochNumber(consensusEpochNum: ConsensusEpochNumber): Unit = {
+    consensusEpochOpt = Some(consensusEpochNum)
+  }
+
+  def updateAccountStateRoot(accountStateRoot: Array[Byte]): Unit = {
+    accountStateRootOpt = Some(accountStateRoot)
+  }
+
+  def updateBaseFee(baseFee: BigInteger): Unit = {
+    baseFeeOpt = Some(baseFee)
+  }
+
+  def getBaseFromFromStorage: Option[BigInteger] = {
+    storage.get(baseFeeKey).asScala.map(wrapper => new BigInteger(wrapper.data))
+  }
+
+  def getBaseFee: BigInteger = {
+    // TODO: default to initial base fee, not zero
+    baseFeeOpt.orElse(getBaseFromFromStorage).getOrElse(BigInteger.ZERO)
+  }
+
+  def setCeased(): Unit = hasCeasedOpt = Some(true)
+
+  // update the database with "dirty" records new values
+  // also increment the height value directly
+  def commit(version: VersionTag): Unit = {
+    val result = Try {
+      saveToStorage(versionToBytes(version))
+    }
+    cleanUpCache()
+    result match {
+      case Failure(exception) =>
+        log.error(s"Error while committing metadata for version $version. The modifications won't be saved", exception)
```

Comment:
at the moment we just log an exception and keep working. This is for sure a end-of-life case.
We need to throw exception if we can't commit.